test/services/oauth2/hooks/index.test.js

```diff
@@ -0,0 +1,88 @@
+const { expect } = require('chai');
+const appPromise = require('../../../../src/app');
+const testObjects = require('../../helpers/testObjects')(appPromise);
+
+const { setIdToken } = require('../../../../src/services/oauth2/hooks/index');
+
+describe.only('oauth2 token test', () => {
+	let app;
+	let server;
+    let createHook;
+
+	before(async () => {
+		app = await appPromise;
+		server = app.listen(0);
+        createHook = (clientId, userId, scopes) => ({
+            app,
+            params: {
+                consentRequest: {
+                    requested_scope: scopes ? scopes : "",
+                    client: {
+                        client_id: clientId
+                    }
+                },
+                account: {
+                    userId: userId
+                },
+                query: {
+                    accept: true
+                },
+                school: {
+                    features: []
+                }
+            },
+            data: {
+                session: {}
+            }
+        });
+	});
+
+	after(() => {
+		server.close();
+	});
+
+    describe('when scope groups is set', () => {
+        it('id_token groups should be defined', async () => {
+            // Arrange
+            const testTeam = await testObjects.createTestTeamWithOwner();
+            const testTool = await testObjects.createTestLtiTool();
+
+            // Act
+            const result = await setIdToken(createHook(testTool.oAuthClientId, testTeam.user._id, "groups"));
+
+            // Assert
+            expect(result).to.not.equal(undefined);
+            expect(result.data.session.id_token.groups).to.not.equal(undefined);
+        });
+
+        it('id_token groups should contain teamid', async () => {
+            // Arrange
+            const testTeam = await testObjects.createTestTeamWithOwner();
+            const testTool = await testObjects.createTestLtiTool();
+
+            // Act
+            const result = await setIdToken(createHook(testTool.oAuthClientId, testTeam.user._id, "groups"));
+
+            // Assert
+            expect(result).to.not.equal(undefined);
```

Comment:
you dont necessarily need this first assertion here, since you already cover it in the other test... but you can keep it if you want. My intention when suggesting multiple "it" statements was to split into one test that covers that you have a valid result (an id token, whatever that means), and one test that checks that there is a group in there. But this is where we get into details...

Reply:
for our younger devs I always suggest trying to use exactly one "expect" statement per test, since that pushes you towards test that test exactly one thing (seperation of concerns). But if you feel you need to, its perfectly viable to first ensure the thing you want to test is actually there... it will also make the error message slightly more readable in case its actually undefined ("expected undefined to not be undefined" vs "cant read property data of undefined"), if that is a concern